system/Validation/StrictRules/Rules.php

```diff
@@ -71,20 +71,28 @@ public function exact_length($str, string $val): bool
     /**
      * Greater than
      *
-     * @param mixed $str
+     * @param mixed $str expects int|string
      */
     public function greater_than($str, string $min): bool
     {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
```

Comment:
Is there a problem with using `@param int|string $str` and `$str = (string) $str;`?

Reply:
`$str` is user input, and we don't know what type it is.

But I found another bug. I'll fix.

Reply:
`$str = (string) $str;` makes `true` `'1'`.
Then `true` will be greater_than `0`. So it is not good.

I pushed a commit to fix another TypeError.

Reply:
The cast is dependent on the docblock type. At this point in dev in 2022 I'm fine requiring users to adhere to docblock types; if we say the parameter is `int|string` and someone passes `bool` then that's on them for not implementing any of the static analysis tools we and the larger PHP community provide.

This works though and is more conservative so it's fine to leave it how you have it.

Reply:
A site user may send unexpected typed values.
In that case, if it is not an attack, returning false is better than TypeError.

Validation rules are special cases. Devs cannot force the input type.
So we cannot set `int|string` in this case.